### DISCUSSION TITLE
Moving HAOS addon InfluxDB database to LXC InfluxDB

### DISCUSSION THREAD

**Discussion by kvmgz** [tteck/Proxmox#1766] [Help Needed] [answered]
Greetins,

I recently moved my HA from a Pi to a ProxmoxVM, now I would like to move further and run InfluxDB (and later Grafana) in LXC.
But how can I migrate my existing InfluxDB database?
If I make a partial backup in Home Assistant of the InfluxDB addon, can that work somehow?
I don't want to lose my collected weather data :S

**Comment by tteck** [comment]
You can try performing a Google search for "how to backup InfluxDB in Home Assistant" to find relevant information on this topic. It may provide you with valuable insights and guidance on how to back up InfluxDB within your Home Assistant setup.

**Reply by kvmgz** [reply]
That's why I turned to you guys, because I couldn't find any practical, useful information :/

**Reply by AlexGeek77** [reply]
If you can't find information it would be great if you document your process. I might be interested in doing the same depending on your results.

**Reply by kvmgz** [reply]
Can you point me in the right direction? I can't find any info for my case, only Docker and other stuff...

**Reply by Vakrehus** [reply]
ssh in to your home assistant, then use this and ignore the docker-part: https://alex3305.github.io/home-assistant-docs/add-ons/influxdb-backup/

**Reply by kvmgz** [reply]
In the end this actually worked, thank you.

**Comment by Vakrehus** [comment]
The backup and restore procedure is quite well documented here: https://docs.influxdata.com/influxdb/v1.8/administration/backup_and_restore/#restore-examples

**Reply by kvmgz** [reply]
Yes, I'm aware of this, just inside Home Assistant's Influx DB I don't have / or know how to / access to it's terminal

**Comment by kvmgz** [comment, marked as answer]
> If you can't find information it would be great if you document your process. I might be interested in doing the same depending on your results.

So my noob step-by-step I was following was this:
Inside Home Assistant I used an "SSH & Terminal" addon. In the addon settings you need to turn off the Protection mode and restart the addon.
With the code below you can find out your InfluxDB container ID:
**echo ] | (sudo docker ps --format "{ {{json .ID}} : {{json .Names}} }" | paste -sd',' && cat) | (echo [ && cat)**

With the next code you can make a backup of the database of your choice:
**sudo docker exec <containerID> influxd backup -portable -database <database name> "/share/influxbackup/database-name"**

Using File Editor addon, with the Enforce Basepath enabled in the configs, you can go to the location where the backup files are.
I couldn't download the whole folder, so I had to download one by one the **.tar** files. There were around 20 in my case, which isn't a big deal.

Now personally I have a drive mounted in OMV and and LXC containers for plex and other -arr apps. So I mounted it into the InfluxDB LXC in it's config file located in the PVE's /etc/pve/local/lxc folder. Nano CONTAINERID.conf added the line **mp0: /mnt/,mp=/mnt/**

I copied the backup files to the desired folder in the mentioned drive.
Inside the InfluxDB LXC console used the command from the documentation to restore the database.
**influxd restore -portable /mnt/FOLDER**

Completed the Influxdb setup to communicate with Home Assistant as in Tteck's guide:
https://www.derekseaman.com/2023/04/home-assistant-installing-influxdb-lxc.html

And now all is working like a charm.
Thanks to all who pointed me in the right direction.
